components/dashboard/src/settings/ProfileInformation.tsx

```diff
@@ -0,0 +1,202 @@
+            return "Please enter a valid email.";
+        }
+        return "";
+    }
+
+    export function NudgeForProfileUpdateModal() {
+        const { user, setUser } = useContext(UserContext);
+        const original = ProfileState.getProfileState(user!);
+        const [profileState, setProfileState] = useState(original);
+        const [errorMessage, setErrorMessage] = useState("");
+        const [visible, setVisible] = useState(shouldNudgeForUpdate(user!));
+
+        const saveProfileState = () => {
+            const error = ProfileState.validate(profileState);
+            setErrorMessage(error);
+            if (error) {
+                return;
+            }
+            const updatedUser = ProfileState.setProfileState(user!, profileState);
+            setUser(updatedUser);
+            getGitpodService().server.updateLoggedInUser(updatedUser);
+            setVisible(shouldNudgeForUpdate(updatedUser!));
+        };
+
+        const cancelProfileUpdate = () => {
+            setProfileState(original);
+            saveProfileState();
+        };
+
+        return (
+            <Modal
+                title="Update Your Profile Information"
+                visible={visible}
+                onClose={cancelProfileUpdate}
+                closeable={true}
+                className="_max-w-xl"
+                buttons={
+                    <div>
+                        <button className="secondary" onClick={cancelProfileUpdate}>
+                            Cancel
+                        </button>
+                        <button onClick={saveProfileState}>Save</button>
+                    </div>
+                }
+            >
+                <ProfileInformation
+                    profileState={profileState}
+                    errorMessage={errorMessage}
+                    updated={false}
+                    setProfileState={setProfileState}
+                />
+            </Modal>
+        );
+    }
+}
+
+export default function ProfileInformation(props: {
+    profileState: ProfileState.ProfileState;
+    setProfileState: (newState: ProfileState.ProfileState) => void;
+    errorMessage: string;
+    updated: boolean;
+    children?: React.ReactChild[] | React.ReactChild;
+}) {
+    return (
+        <div>
+            <h3>Profile</h3>
+            <p className="text-base text-gray-500 pb-4 max-w-2xl">
+                The following information will be used to set up Git configuration. You can override Git author name and
+                email per project by using the default environment variables <CodeText>GIT_AUTHOR_NAME</CodeText>,{" "}
+                <CodeText>GIT_COMMITTER_NAME</CodeText>, <CodeText>GIT_AUTHOR_EMAIL</CodeText> and{" "}
+                <CodeText>GIT_COMMITTER_EMAIL</CodeText>.
+            </p>
+            {props.errorMessage.length > 0 ? (
+                <div className="dark:bg-gray-800 bg-gitpod-kumquat-light rounded-md p-3 text-gitpod-red text-sm mb-2">
```

Comment:
**nitpick**: Minor layout change for adding visual balance, based on previous work on defining breakpoints, etc, see relevant comment in https://github.com/gitpod-io/gitpod/issues/8133#issuecomment-1039013543 for more context.

```suggestion
                <div className="dark:bg-gray-800 bg-gitpod-kumquat-light rounded-md p-3 text-gitpod-red text-sm mb-2 max-w-xl">
```

| BEFORE | AFTER |
|-|-|
| <img width="1440" alt="Screenshot 2022-06-30 at 6 25 15 PM" src="https://user-images.githubusercontent.com/120486/176716128-ed83d82e-f88c-4618-845c-717bb5551c8b.png"> | <img width="1440" alt="Screenshot 2022-06-30 at 6 25 24 PM" src="https://user-images.githubusercontent.com/120486/176716132-9e6a2e3c-bf2a-4759-9714-df28fdae2729.png">  |

Reply:
applied suggestion as well as moved to <Alert />

Reply:
💯